engine/src/conversion/apivec.rs

```diff
@@ -89,6 +89,10 @@ impl<P: AnalysisPhase> ApiVec<P> {
         self.apis.iter()
     }
 
+    pub(crate) fn iter_mut(&mut self) -> impl Iterator<Item = &mut Api<P>> {
+        self.apis.iter_mut()
```

Comment:
Sorry, we can't do this. In fact I worked really hard to remove this in bcfeb6fc85f5194d310633cb752123dc567c620d! The problem is that it might invalidate the invariants which `ApiVec` promises to maintain.

I struggled for a while with a custom iterator which then re-checked the invariants on its `Drop`, but borrow checker fights caused me to give up, and rewrite calling code so it didn't need `iter_mut`. You're welcome to try that again.

Reply:
WDYT about `fn mutate_fun_analysis<F: Fn(&mut P::FunAnalysis)>(&mut self, f: F)` (and another one for structs)? Kind of ugly, but those don't allow changing the names, so the invariants can't be invalidated.

Reply:
Or, a bit less ugly but with a runtime assertion, `fn filter_inplace<F: Fn(&mut Api)>(&mut self, f: F)` which `assert!`s that the name doesn't change after each function call.

Reply:
Either would be fine with me, yes. I think on the whole I prefer the former, but I don't mind!

Reply:
Done